### DISCUSSION TITLE
Installing module in a location without a parent shelf...

### DISCUSSION THREAD

**Discussion by shatt79** [netbox-community/netbox#10467] [General]
We have a room at our facility to store spare equipment.  In Netbox, we have this room built as a location within a site.  Has anyone figured out a way to "install" modules as unracked equipment in a location?  I'd like to be able to keep an inventory of spare equipment.  This works just fine for devices, but not for modules.

**Comment by candlerb** [comment]
There are "inventory items" and now more recently "modules".  I looked into this before for inventory items.

Originally you couldn't move inventory items from one device to another, but now you can, so it's possible to have a dummy container device for them.

The same might be true for modules, but I haven't checked this.

**Reply by shatt79** [reply]
Correct, but I'm trying to figure out if there is any way for a module to be placed in a location without a parent "dummy" device.

**Reply by candlerb** [reply]
There isn't - the data model doesn't allow it.

```
class Module(NetBoxModel, ConfigContextModel):
    """
    A Module represents a field-installable component within a Device which may itself hold multiple device components
    (for example, a line card within a chassis switch). Modules are instantiated from ModuleTypes.
    """
    device = models.ForeignKey(
        to='dcim.Device',
        on_delete=models.CASCADE,
        related_name='modules'
    )
```

The `device` field on a module is not nullable, so it cannot exist without a device.

**Reply by shatt79** [reply]
I figured as much.  I just wasn't sure if there was any kind of magical workaround.  Thanks.

**Reply by cyberndj** [reply]
Howdy,

As a thought, you could create a generic chassis device type with lots of module bays. Then you could just move the module to the generic chassis as a generic location to hold the modules.

**Reply by candlerb** [reply]
Looks like as of Netbox v3.3.5 this is no longer possible:

> * [#10513](https://github.com/netbox-community/netbox/issues/10513) - Disable the reassignment of a module to a new device
